cpp/include/raft/spatial/knn/detail/ball_cover.cuh

```diff
@@ -340,6 +363,16 @@ void rbc_all_knn_query(const raft::handle_t& handle,
   rmm::device_uvector<value_idx> R_knn_inds(k * index.m, handle.get_stream());
   rmm::device_uvector<value_t> R_knn_dists(k * index.m, handle.get_stream());
 
+  // Initialize the uvectors
+  thrust::fill(handle.get_thrust_policy(),
+               R_knn_inds.begin(),
+               R_knn_inds.end(),
+               std::numeric_limits<value_idx>::max());
+  thrust::fill(handle.get_thrust_policy(),
+               R_knn_dists.begin(),
+               R_knn_dists.end(),
+               std::numeric_limits<value_t>::max());
+
   // For debugging / verification. Remove before releasing
   rmm::device_uvector<value_int> dists_counter(index.m, handle.get_stream());
```

Comment:
While we are here, should this be removed? It seems, the `dists_counter` and `post_dists_counter` are not used by underline kernels.

Reply:
I've been using them for debugging (this algorithm is slow going and still isn't quite "finished").